conda_smithy/linter/lints.py

```diff
@@ -796,8 +796,15 @@ def flatten_reqs(reqs):
     # this check needs to be done per output --> use separate (unflattened) requirements
     for build_reqs in all_build_reqs:
         has_compiler = any(pat_compiler_stub.match(rq) for rq in build_reqs)
-        stdlib_stub = "c_stdlib_stub" if recipe_version == 0 else "${{ stdlib"
-        if has_compiler and stdlib_stub not in build_reqs:
+        stdlib_regex = (
+            # we need the C stdlib, not just any invocation of the stdlib jinja
+            "c_stdlib_stub"
+            if recipe_version == 0
+            else r"\$\{\{ stdlib\(['\"]c['\"]"
```

Comment:
Won't this mis the original error for v1 recipes where the c stdlib was <something>_c?

Reply:
I don't understand your question. By v1 you mean the new format, right? For that, we're only changing the search from the literal `"${{ stdlib"` to the regex that also ensures the `("c"` part, which you had asked for.

I don't know where `something_c` would ever come into play, either for v0 or v1 (there's `c_stdlib_stub` in v0, which I'm using here).

Reply:
The linter originally failed because it didn't recognize m2w64_c as a valid stdlib. In the v1 recipe this would be "${{ stdlib('m2w64_c') }}". The code was written would fail to recognize this stdlib and thus have the same bug, but now on v1 recipes.

Reply:
yeah... that one's on me. I forgot the first part of the reason of this PR when I picked it up again. sorry